src/main/java/com/example/demo/DemoApplication.java

```diff
@@ -2,8 +2,12 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@EntityScan(basePackages = {"com.example"})
+@ComponentScan(basePackages = {"com.example"})
```

Comment:
why not invent something more descriptive than `com.example` 
Also I'm not sure you need them at all, if all of your classes are under the same package

Reply:
Without it, my application wont work properly. In console there is no hibernate requests, and any request to application has 404 not found response: {
    "timestamp": "2024-04-20T08:10:45.536+00:00",
    "status": 404,
    "error": "Not Found",
    "path": "/api/books"
}